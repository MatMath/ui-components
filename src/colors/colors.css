:root {

/* -- DASH_GREEN -- */
--dash-green-00: #0E353D;
--dash-green-01: #3E5D64;
--dash-green-02: #6E868B;
--dash-green-03: #9FAEB1;
--dash-green-04: #CFD7D8;
--dash-green-05: #D9DFE0;
--dash-green-06: #F5F7F7;

/* -- DASH_DARKER_GREEN -- */
--dash-darker-green-00: #071A1E;

/* -- MID_GREEN -- */
--mid-green-00: #0E6476;
--mid-green-01: #3E8391;
--mid-green-02: #6EA2AD;
--mid-green-03: #9FC1C8;
--mid-green-04: #CFE0E4;
--mid-green-05: #D9E6E9;

/* -- PINK -- */
--pink-00: #FDB5AF;
--pink-01: #FDC4BF;
--pink-02: #FED3CF;
--pink-03: #FEE1DF;
--pink-04: #FFF0EF;
--pink-05: #FFF3F2;

/* -- ORANGE -- */
--orange-00: #FE5B33;
--orange-01: #FE7C5C;
--orange-02: #FE9D85;
--orange-03: #FFBDAD;
--orange-04: #FFDED6;
--orange-05: #FFE5DE;

/* -- FUNCTIONAL_RED -- */
--functional-red-00: #4E000D;
--functional-red-01: #8B0018;
--functional-red-02: #FF002D;
--functional-red-03: #FFA49F;

/* -- RED -- */
--red-00: #BF0A33;
--red-01: #CC3B5C;
--red-02: #D96C85;
--red-03: #E59DAD;
--red-04: #F2CED6;
--red-05: #F5D8DE;

/* -- GREY -- */
--grey-00: #615B57;
--grey-01: #999592;
--grey-02: #CCC7C2;
--grey-03: #D6D2CE;
--grey-04: #E0DDDA;
--grey-05: #EBE9E7;
--grey-06: #F5F4F3;

/* -- BLACK -- */
--black: #000000;

/* -- WHITE -- */
--white: #FFFFFF;

/* -- VALIDATOR_GREEN -- */
--validator-green: #20B422;

/* -- TRANSPARENT -- */
--transparent: transparent;
}
