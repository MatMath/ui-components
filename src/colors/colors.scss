/* -- DASH_GREEN -- */
$dash-green-00: #0e353d;
$dash-green-01: #3e5d64;
$dash-green-02: #6e868b;
$dash-green-03: #9faeb1;
$dash-green-04: #cfd7d8;
$dash-green-05: #d9dfe0;
$dash-green-06: #f5f7f7;

/* -- DASH_DARKER_GREEN -- */
$dash-darker-green-00: #071a1e;

/* -- MID_GREEN -- */
$mid-green-00: #0e6476;
$mid-green-01: #3e8391;
$mid-green-02: #6ea2ad;
$mid-green-03: #9fc1c8;
$mid-green-04: #cfe0e4;
$mid-green-05: #d9e6e9;

/* -- PINK -- */
$pink-00: #fdb5af;
$pink-01: #fdc4bf;
$pink-02: #fed3cf;
$pink-03: #fee1df;
$pink-04: #fff0ef;
$pink-05: #fff3f2;

/* -- ORANGE -- */
$orange-00: #fe5b33;
$orange-01: #fe7c5c;
$orange-02: #fe9d85;
$orange-03: #ffbdad;
$orange-04: #ffded6;
$orange-05: #ffe5de;

/* -- FUNCTIONAL_RED -- */
$functional-red-00: #4e000d;
$functional-red-01: #8b0018;
$functional-red-02: #ff002d;
$functional-red-03: #ffa49f;

/* -- RED -- */
$red-00: #bf0a33;
$red-01: #cc3b5c;
$red-02: #d96c85;
$red-03: #e59dad;
$red-04: #f2ced6;
$red-05: #f5d8de;

/* -- GREY -- */
$grey-00: #615b57;
$grey-01: #999592;
$grey-02: #ccc7c2;
$grey-03: #d6d2ce;
$grey-04: #e0ddda;
$grey-05: #ebe9e7;
$grey-06: #f5f4f3;

/* -- BLACK -- */
$black: #000;

/* -- WHITE -- */
$white: #fff;

/* -- VALIDATOR_GREEN -- */
$validator-green: #20b422;

/* -- ACCESSIBLE_VALIDATOR_GREEN -- */
$accessible-validator-green: #278637;

/* -- ACCESSIBLE_VALIDATOR_RED -- */
$accessible-validator-red: #e00635;

/* -- TRANSPARENT -- */
$transparent: transparent;
