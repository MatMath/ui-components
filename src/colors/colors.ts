/* -- DASH_GREEN -- */
export const dashGreen00 = '#0E353D';
export const dashGreen01 = '#3E5D64';
export const dashGreen02 = '#6E868B';
export const dashGreen03 = '#9FAEB1';
export const dashGreen04 = '#CFD7D8';
export const dashGreen05 = '#D9DFE0';
export const dashGreen06 = '#F5F7F7';

/* -- DASH_DARKER_GREEN -- */
export const dashDarkerGreen00 = '#071A1E';

/* -- MID_GREEN -- */
export const midGreen00 = '#0E6476';
export const midGreen01 = '#3E8391';
export const midGreen02 = '#6EA2AD';
export const midGreen03 = '#9FC1C8';
export const midGreen04 = '#CFE0E4';
export const midGreen05 = '#D9E6E9';

/* -- PINK -- */
export const pink00 = '#FDB5AF';
export const pink01 = '#FDC4BF';
export const pink02 = '#FED3CF';
export const pink03 = '#FEE1DF';
export const pink04 = '#FFF0EF';
export const pink05 = '#FFF3F2';

/* -- ORANGE -- */
export const orange00 = '#FE5B33';
export const orange01 = '#FE7C5C';
export const orange02 = '#FE9D85';
export const orange03 = '#FFBDAD';
export const orange04 = '#FFDED6';
export const orange05 = '#FFE5DE';

/* -- FUNCTIONAL_RED -- */
export const functionalRed00 = '#4E000D';
export const functionalRed01 = '#8B0018';
export const functionalRed02 = '#FF002D';
export const functionalRed03 = '#FFA49F';

/* -- RED -- */
export const red00 = '#BF0A33';
export const red01 = '#CC3B5C';
export const red02 = '#D96C85';
export const red03 = '#E59DAD';
export const red04 = '#F2CED6';
export const red05 = '#F5D8DE';

/* -- GREY -- */
export const grey00 = '#615B57';
export const grey01 = '#999592';
export const grey02 = '#CCC7C2';
export const grey03 = '#D6D2CE';
export const grey04 = '#E0DDDA';
export const grey05 = '#EBE9E7';
export const grey06 = '#F5F4F3';

/* -- BLACK -- */
export const black = '#000000';

/* -- WHITE -- */
export const white = '#FFFFFF';

/* -- VALIDATOR_GREEN -- */
export const validatorGreen = '#20B422';

/* -- ACCESSIBLE_VALIDATOR_GREEN -- */
export const accessibleValidatorGreen = '#278637';

/* -- ACCESSIBLE_VALIDATOR_RED -- */
export const accessibleValidatorRed = '#e00635';

/* -- TRANSPARENT -- */
export const transparent = 'transparent';
