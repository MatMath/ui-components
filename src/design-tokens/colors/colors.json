{
  "dash-green": {
    "00": "#0E353D",
    "01": "#3E5D64",
    "02": "#6E868B",
    "03": "#9FAEB1",
    "04": "#CFD7D8",
    "05": "#D9DFE0",
    "06": "#F5F7F7"
  },
  "dash-darker-green": {
    "00": "#071A1E"
  },
  "mid-green": {
    "00": "#0E6476",
    "01": "#3E8391",
    "02": "#6EA2AD",
    "03": "#9FC1C8",
    "04": "#CFE0E4",
    "05": "#D9E6E9"
  },
  "pink": {
    "00": "#FDB5AF",
    "01": "#FDC4BF",
    "02": "#FED3CF",
    "03": "#FEE1DF",
    "04": "#FFF0EF",
    "05": "#FFF3F2"
  },
  "orange": {
    "00": "#FE5B33",
    "01": "#FE7C5C",
    "02": "#FE9D85",
    "03": "#FFBDAD",
    "04": "#FFDED6",
    "05": "#FFE5DE"
  },
  "functional-red": {
    "00": "#4E000D",
    "01": "#8B0018",
    "02": "#FF002D",
    "03": "#FFA49F"
  },
  "red": {
    "00": "#BF0A33",
    "01": "#CC3B5C",
    "02": "#D96C85",
    "03": "#E59DAD",
    "04": "#F2CED6",
    "05": "#F5D8DE"
  },
  "warning": {
    "00": "#BF2600"
  },
  "grey": {
    "00": "#615B57",
    "01": "#999592",
    "02": "#CCC7C2",
    "03": "#D6D2CE",
    "04": "#E0DDDA",
    "05": "#EBE9E7",
    "06": "#F5F4F3"
  },
  "black": "#000000",
  "white": "#FFFFFF",
  "validator-green": "#20B422",
  "accessible-validator-green": "#278637",
  "accessible-validator-red": "#e00635",
  "transparent": "transparent"
}
