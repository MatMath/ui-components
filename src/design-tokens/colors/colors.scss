/* -- PRIMARY -- */
$primary-100: #cce5ea;
$primary-300: #8cc4d0;
$primary-500: #007c97;
$primary-600: #005d71;
$primary-700: #003945;
$primary-800: #002e38;
$primary-900: #062931;

/* -- SECONDARY -- */
$secondary-300: #ffc1a3;
$secondary-500: #ff5412;
$secondary-700: #b34015;
$secondary-900: #823214;

/* -- SUCCESS -- */
$success-300: #c1eac1;
$success-500: #20b422;
$success-700: #157917;
$success-900: #105c11;

/* -- DANGER -- */
$danger-300: #ffe3dd;
$danger-500: #ff002d;
$danger-700: #8b0018;
$danger-900: #4e000d;

/* -- WARNING -- */
$warning-300: #fffac1;
$warning-500: #ffe900;
$warning-700: #595200;
$warning-900: #2e2a00;

/* -- INFO -- */
$info-300: #bdedff;
$info-500: #00b8ff;
$info-700: #007dad;
$info-900: #00e042;

/* -- LIGHT -- */
$light-100: #ffffff;
$light-200: #fafafa;
$light-300: #f6f6f6;
$light-400: #e7e7e7;
$light-500: #d6d6d6;
$light-600: #9b9b9b;
$light-700: #474747;
$light-900: #191919;
