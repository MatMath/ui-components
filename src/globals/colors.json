{
  "primary": {
    "100": "rgb(204,229,234)",
    "300": "rgb(140,196,208)",
    "500": "rgb(0,124,151)",
    "600": "rgb(0,93,113)",
    "700": "rgb(0,57,69)",
    "800": "rgb(0,46,56)",
    "900": "rgb(6,41,49)"
  },
  "secondary": {
    "300": "rgb(255,193,163)",
    "500": "rgb(255,84,18)",
    "700": "rgb(179,64,21)"
  },
  "success": {
    "300": "rgb(193,234,193)",
    "500": "rgb(32,180,34)",
    "700": "rgb(21,121,23)"
  },
  "danger": {
    "300": "rgb(255,227,221)",
    "500": "rgb(255,0,45)",
    "700": "rgb(139,0,24)"
  },
  "warning": {
    "300": "rgb(255,250,193)",
    "500": "rgb(255,233,0)",
    "700": "rgb(89,82,0)"
  },
  "info": {
    "300": "rgb(153,226,254)",
    "500": "rgb(0,184,255)",
    "700": "rgb(0,125,173)",
    "900": "rgb(0,48,66)"
  },
  "gray": {
    "100": "rgb(255,255,255)",
    "200": "rgb(250,250,250)",
    "300": "rgb(246,246,246)",
    "400": "rgb(231,231,231)",
    "500": "rgb(214,214,214)",
    "600": "rgb(155,154,155)",
    "700": "rgb(71,71,71)",
    "900": "rgb(25,25,25)"
  }
}