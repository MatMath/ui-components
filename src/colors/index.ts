import { BasicColor, ColorExclude } from '@typings/Color.d';
import * as colors from '@design-tokens/colors.json';

export const $primary: ColorExclude<'200' | '400'> = colors.primary;

export const $secondary: BasicColor = colors.secondary;

export const $success: BasicColor = colors.success;

export const $danger: BasicColor = colors.danger;

export const $warning: BasicColor = colors.warning;

export const $info: BasicColor = colors.info;

export const $light: ColorExclude<'800'> = colors.light;

export const $transparent = 'transparent';
