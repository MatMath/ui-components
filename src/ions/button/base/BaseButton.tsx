import * as React from 'react';
import styled, { DefaultTheme } from 'styled-components';
import { $light, $transparent } from '@globals/colors';
import { Text } from '@ions';
import { BaseButtonProps } from '@atoms/button/types';

type Theme = {
  [key in keyof DefaultTheme]: string;
};

export interface BaseProps extends BaseButtonProps {
  children?: React.ReactNode;
  theme: Theme;
}

const getActiveColor = ({ theme }: BaseProps): string => theme.activeBackground;

const Button = styled.button<BaseProps>`
  color: ${({ theme }): string => theme.color || $light[100]};
  background-color: ${({ theme }): string => theme.background};
  opacity: ${({ disabled }): number => (disabled ? 0.2 : 1)}
  cursor: ${({ disabled }): string => (disabled ? 'not-allowed' : 'pointer')}
  border: 1px solid ${$transparent};
  transition: all 0.2s ease-in;
  border-radius: 4px;  
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: inline-flex;
  position: relative;
  align-items: center;
  justify-content: center;
  outline: 0;
  &:hover, &:active {
    border: 1px solid ${({ theme }): string => theme.hoverBorder}
  }
  &:active {
    background-color: ${getActiveColor};
    border-color: ${getActiveColor}
    color: ${({ theme }): string =>
      theme.color !== $light[700] ? $light[100] : theme.color};
  }
  &:disabled span {
    pointer-events: none;
  }
`;

const BaseButton: React.FunctionComponent<BaseProps> = ({
  children,
  ...props
}: BaseProps) => (
  <Button {...props}>
    <Text>{children || 'Default'}</Text>
  </Button>
);

export default BaseButton;
