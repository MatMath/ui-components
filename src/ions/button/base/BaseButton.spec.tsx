import * as React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import BaseButton from './BaseButton';
import { $light, $primary } from '@globals/colors';

const defaultTheme = {
  color: $primary[300],
  background: $primary[500],
  hoverBorder: $primary[600],
  activeBackground: $primary[700]
};

describe('<BaseButton>', () => {
  describe('global render', () => {
    it('should render', () => {
      const Button = mount(<BaseButton theme={defaultTheme} />);

      expect(Button).toMatchSnapshot();
    });
  });

  describe('style properties', () => {
    it('should render with the specified theme', () => {
      const Button = mount(<BaseButton theme={defaultTheme} />);

      expect(Button).toHaveStyleRule('color', defaultTheme.color);
      expect(Button).toHaveStyleRule(
        'background-color',
        defaultTheme.background
      );
    });

    it('should render default color if none is provided', () => {
      const { color, ...theme } = defaultTheme;
      const Button = mount(<BaseButton theme={theme} />);

      expect(Button).toHaveStyleRule('color', $light[100]);
    });

    it('should have the correct style when disabled', () => {
      const Button = mount(<BaseButton theme={defaultTheme} disabled />);

      expect(Button).toHaveStyleRule('opacity', '0.2');
      expect(Button).toHaveStyleRule('cursor', 'not-allowed');
    });
  });

  describe('children render', () => {
    it('should render Default when no children are specified', () => {
      const Button = mount(<BaseButton theme={defaultTheme} />);

      expect(Button.text()).toBe('Default');
    });

    it('should render the specified text', () => {
      const text = 'Hello There';
      const Button = mount(
        <BaseButton theme={defaultTheme}>{text}</BaseButton>
      );

      expect(Button.text()).toEqual(text);
    });

    it('should render the specified children', () => {
      const Button = mount(
        <BaseButton theme={defaultTheme}>
          <label>Ok</label>
        </BaseButton>
      );

      expect(Button.find('label')).toBeTruthy();
      expect(Button.text()).toEqual('Ok');
    });
  });

  describe('interaction', () => {
    it('should execute action on click', () => {
      const Button = mount(
        <BaseButton
          theme={defaultTheme}
          onClick={() => window.alert('hello')}
        />
      );

      window.alert = jest.fn();
      Button.simulate('click');
      expect(window.alert).toHaveBeenCalledTimes(1);
      expect(window.alert).toHaveBeenCalledWith('hello');
    });
  });
});
