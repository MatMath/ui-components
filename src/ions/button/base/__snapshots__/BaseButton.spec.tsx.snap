// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton> global render should render 1`] = `
.c1 {
  line-height: 1.75;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
}

.c0 {
  color: #8cc4d0;
  background-color: #007c97;
  opacity: 1;
  cursor: pointer;
  border: 1px solid transparent;
  -webkit-transition: all 0.2s ease-in;
  transition: all 0.2s ease-in;
  border-radius: 4px;
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: 0;
}

.c0:hover,
.c0:active {
  border: 1px solid #005d71;
}

.c0:active {
  background-color: #003945;
  border-color: #003945;
  color: #ffffff;
}

.c0:disabled span {
  pointer-events: none;
}

<BaseButton
  theme={
    Object {
      "activeBackground": "#003945",
      "background": "#007c97",
      "color": "#8cc4d0",
      "hoverBorder": "#005d71",
    }
  }
>
  <styled.button
    theme={
      Object {
        "activeBackground": "#003945",
        "background": "#007c97",
        "color": "#8cc4d0",
        "hoverBorder": "#005d71",
      }
    }
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  opacity: ",
              [Function],
              "
  cursor: ",
              [Function],
              "
  border: 1px solid ",
              "transparent",
              ";
  transition: all 0.2s ease-in;
  border-radius: 4px;  
  line-height: 1.75;
  padding: 6px 16px;
  min-width: 64px;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  display: inline-flex;
  position: relative;
  align-items: center;
  justify-content: center;
  outline: 0;
  &:hover, &:active {
    border: 1px solid ",
              [Function],
              "
  }
  &:active {
    background-color: ",
              [Function],
              ";
    border-color: ",
              [Function],
              "
    color: ",
              [Function],
              ";
  }
  &:disabled span {
    pointer-events: none;
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      theme={
        Object {
          "activeBackground": "#003945",
          "background": "#007c97",
          "color": "#8cc4d0",
          "hoverBorder": "#005d71",
        }
      }
    >
      <button
        className="c0"
      >
        <styled.span>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  line-height: 1.75;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <span
              className="c1"
            >
              Default
            </span>
          </StyledComponent>
        </styled.span>
      </button>
    </StyledComponent>
  </styled.button>
</BaseButton>
`;
